benchmarks/nel/scripts/datasets/evaluation.py

```diff
@@ -0,0 +1,207 @@
+            self.false_pos += not candidate_is_correct
+
+    def calculate_precision(self):
+        if self.true_pos == 0:
+            return 0.0
+        else:
+            return self.true_pos / (self.true_pos + self.false_pos)
+
+    def calculate_recall(self):
+        if self.true_pos == 0:
+            return 0.0
+        else:
+            return self.true_pos / (self.true_pos + self.false_neg)
+
+    def calculate_fscore(self):
+        p = self.calculate_precision()
+        r = self.calculate_recall()
+        if p + r == 0:
+            return 0.0
+        else:
+            return 2 * p * r / (p + r)
+
+
+class EvaluationResults(object):
+    def __init__(self, name: str):
+        self.name = name
+        self.metrics = Metrics()
+        self.metrics_by_label = defaultdict(Metrics)
+
+    def update_metrics(self, ent_label: str, true_ent_kb_id_: str, cand_kb_ids_: Set[str]) -> None:
+        """ Update metrics over all candidate labels and per candidate labels.
+        ent_label (str): Recognized entity label.
+        true_ent_kb_id_ (str): True entity's external KB ID.
+        cand_kb_ids_ (Set[str]): Set of candidates' external KB IDs.
+        """
+        self.metrics.update_results(true_ent_kb_id_, cand_kb_ids_)
+        self.metrics_by_label[ent_label].update_results(true_ent_kb_id_, cand_kb_ids_)
+
+    def _extend_report_table(self, table: prettytable.PrettyTable, labels: Tuple[str, ...]) -> None:
+        """ Extend existing PrettyTable with collected metrics.
+        model_name (str): Model name.
+        table (prettytable.PrettyTable): PrettyTable object for evaluation results.
+        labels (Tuple[str, ...]): Labels in sequence to be added to table.
+        """
+        table.add_row([
+            self.name.title(),
+            str(self.metrics.true_pos),
+            str(self.metrics.false_pos),
+            str(self.metrics.false_neg),
+            f"{round(self.metrics.calculate_fscore(), 3)}",
+            f"{round(self.metrics.calculate_recall(), 3)}",
+            f"{round(self.metrics.calculate_precision(), 3)}",
+            *[self.metrics_by_label[label].calculate_fscore() for label in labels]
+        ])
+
+    @staticmethod
+    def report(evaluation_results: Tuple["EvaluationResults"]) -> None:
+        """ Reports evaluation results.
+        evaluation_result (Tuple["EvaluationResults"]): Evaluation results.
+        """
+        labels = sorted(list({label for eval_res in evaluation_results for label in eval_res.metrics_by_label}))
+        table = prettytable.PrettyTable(
+            field_names=[
+                "model_name", "TPOS", "FPOS", "FNEG", "F-score", "Recall", "Precision",
+                *[f"F-score ({label})" for label in labels]
+            ]
+        )
+
+        for eval_result in evaluation_results:
+            eval_result._extend_report_table(table, tuple(labels))
+
+        print(table)
```

Comment:
You need a pretty wide screen to be able to read this on the console.

I'd suggest printing the first bit separately (TPOS, FPOS, FNEG, F-SCORE, Recall, Precision) and then make a separate table which has each NE type on a separate row and for each you could print F/r/p